app/uk/gov/hmrc/customs/inventorylinking/export/controllers/actionbuilders/AuthAction.scala

```diff
@@ -92,18 +90,31 @@ class AuthAction @Inject()(
     }
   }
 
-  private def eitherSubmitterIdentifierWithValidationCSP[A](implicit vhr: ValidatedHeadersRequest[A]) = {
+  private def eitherSubmitterIdentifierWithValidationCSP[A](implicit asf: ApiSubscriptionFieldsRequest[A]) = {
     val maybeSubmitterId: Option[String] = maybeHeaderCaseInsensitive(XSubmitterIdentifierHeaderName)
-    maybeValidEori(maybeSubmitterId).fold[Either[Result, Eori]] {
-      logger.error(s"Submitter identifier invalid or not present for CSP ($maybeSubmitterId)")
-      Left(errorResponseSubmitterIdentifierHeaderMissing.XmlResult.withConversationId)
-    } { eori =>
-      logger.debug("Authorising as CSP")
-      Right(eori)
+    maybeSubmitterId.fold[Either[Result, Eori]] {
+      asf.apiSubscriptionFields.fields.authenticatedEori.fold[Either[Result, Eori]] {
+        logger.error(s"Submitter identifier and Authenticated EORI not present for CSP")
+        Left(ErrorInternalServerError.XmlResult.withConversationId)
+      } { eoriFromFields =>
+        if (eoriFromFields.trim.nonEmpty) {
+          Right(Eori(eoriFromFields))
+        } else {
+          logger.error(s"Authenticated EORI ($eoriFromFields) invalid for CSP")
+          Left(ErrorInternalServerError.XmlResult.withConversationId)
+        }
+      }
+    }{eoriFromHeader =>
+      if (validEori(eoriFromHeader)) {
+        Right(Eori(eoriFromHeader))
+      } else {
+        logger.error(s"Submitter identifier ($eoriFromHeader) invalid for CSP")
```

Comment:
Similar to above, would `missing or invalid` rather than `invalid` be a better formulation here?